autonomy_core/map_plan/mapper/src/test_voxel_mapper.cpp

```diff
@@ -0,0 +1,428 @@
+VoxelMapperTest::VoxelMapperTest()
+    :VoxelMapperTest(-100, -100, -5, 200, 200, 10, 0.5, 30) {}
+
+VoxelMapperTest::VoxelMapperTest(double x_orig, double y_orig, double z_orig,
+                                 int x_dim, int y_dim, int z_dim, double res, int decay_times)
+    : x_origin_(x_orig), y_origin_(y_orig), z_origin_(z_orig),
+      x_dim_(x_dim), y_dim_(y_dim), z_dim_(z_dim),
+      resolution_(res), decay_times_(decay_times)
+{
+    const Eigen::Vector3d origin(x_origin_, y_origin_, z_origin_);
+    const Eigen::Vector3d dimensions(x_dim_, y_dim_, z_dim_);
+    p_test_mapper_.reset(new mapper::VoxelMapper(origin, dimensions, resolution_, 0, decay_times_));
+}
+
+
+/*** Defining Tests ***/
+
+/**
+ * @brief The allocate() method is able to relocate the current map into another part of the
+ * world, but this functionality is currently NOT being used since allocate() is only called
+ * in the constructor of the VoxelMapper class. The map is stored in the map_ data member and
+ * we could access it directly via friendship to verify the correct initialization of allocate(),
+ * but this test should not depend on the data type of the map_ variable; for this reason, we use
+ * the getMap() method to check the status of the map. In the first three assertions, we check that
+ * the map is created with the correct dimensions. In the second portion we check that relocating is
+ * working properly.
+ */
+TEST_F(VoxelMapperTest, TestAllocate)
+{
+    /*****  PART1: Test that map is created with correct dimensions  *****/
+    // Creating voxel mapper object with same parameters as global_voxel_mapper in source file:
+    // local_global_mapper.cpp
+    Eigen::Vector3d origin(-100, -100, -5);
+    Eigen::Vector3d dimensions(200, 200, 10);
+
+    time_start = std::chrono::steady_clock::now();
+    p_test_mapper_.reset(new mapper::VoxelMapper(origin, dimensions, 0.5, 0, 30));
+    time_end = std::chrono::steady_clock::now();
+    std::cout << "CALLBACK Duration (Constructor): " <<
+        std::chrono::duration_cast<std::chrono::microseconds>(time_end - time_start).count()
+        << "us" << std::endl;
+
+    EXPECT_EQ(p_test_mapper_->getMap().data.size(), 3200000);
+
+    // Creating voxel mapper object with same parameters as storage_voxel_mapper in source file:
+    // local_global_mapper.cpp
+    origin = Eigen::Vector3d(-100, -100, -5);
+    dimensions = Eigen::Vector3d(200, 200, 12);
+    p_test_mapper_.reset(new mapper::VoxelMapper(origin, dimensions, 0.25, 0, 30));
+    EXPECT_EQ(p_test_mapper_->getMap().data.size(), 30720000);
+
+    // Creating voxel mapper object with custom parameters
+    origin = Eigen::Vector3d(-5, -5, -5);
+    dimensions = Eigen::Vector3d(10, 10, 10);
+    p_test_mapper_.reset(new mapper::VoxelMapper(origin, dimensions, 0.1, 0, 30));
+    EXPECT_EQ(p_test_mapper_->getMap().data.size(), 1000000);
+
+
+    /*****  PART2: Now test whether the relocation is working  *****/
```

Comment:
Do not have multiple parts in a unit test, make a separate test.

Reply:
Done, changing now